enhancements/security/confidential-clusters.md

```diff
@@ -0,0 +1,619 @@
+
+* Provide a seamless integration of confidential computing and remote
+  attestation from the cluster admin perspective.
+
+* Support cluster upgrades and other lifecycle operations while preserving
+  cluster confidentiality.
+
+### Non-Goals
+
+* This enhancement does not aim to protect from a malicious cluster operator or
+  from an attacker that managed to elevate their privileges to cluster admin.
+
+* This enhancement does not aim to provide data encryption outside of the
+  confidential computing environment (for example network encryption, additional
+  disk encryption), though existing OpenShift mechanisms to do that are
+  available.
+
+* This enhancement does not cover changes to application-level data
+  encryption. It focuses on protecting data in memory and on the disk at the
+  infrastructure layer.
+
+* This enhancement does not address the security of the underlying cloud
+  provider's hardware or hypervisor outside of the CVM's confidential execution
+  environment.
+
+## Proposal
+
+Run all OpenShift nodes on Confidential VMs (CVMs). Use remote attestation to
+verify the integrity and authenticity of a new node's hardware and software
+before sharing secrets with that node and allowing it to join the cluster.
+
+This implementation will happen in two phases.
+
+* In the first phase, we will consider the bootstrap node and the first boot of
+  each new node to be trusted. In this phase, only the confidentiality of the
+  cluster will be guaranteed. We will assume the attacker can read data but not
+  write data (to the disk, cloud metadata config, etc.).
+
+* In the second phase, we will remove the need to trust the bootstrap node and
+  the first boot of each node. Once completed, both confidentiality and
+  integrity will be guaranteed.
+
+We are working on a more detailed threat model, which will be submitted in a
+later stage.
+
+In the first phase (confidentiality), the following changes are needed to those
+components:
+
+* OpenShift API
+  * Allow nodes to be marked as confidential. This is specific per cloud
+    provider and per Hardware manufacturer.
+  * Request/Instruct cloud providers to run nodes as CVMs.
+
+* Installer
+  * Allow users to specify they want to run OpenShift as a Confidential Cluster
+    (cloud provider specific).
+  * Deploy the Confidential Cluster Operator on the bootstrap node
+
+* Confidential Cluster Operator
+  * Setup a Trustee (attestation service) instance in the cluster to attest
+    nodes.
+  * Setup attestation and resource access policies in Trustee.
+  * Provide a registration server for new nodes to trigger the provisioning of
+    secrets.
+  * Setup a MachineConfig to instruct new nodes to attest themselves.
+  * Watch for cluster or OS image updates, compute and update the set of
+    reference-values (expected "correct" values) in Trustee.
+
+* RHEL CoreOS
+  * Add support for composefs (native), UKI, and systemd-boot to bootc (Bootable
```

Comment:
Since the MCO and on-cluster RHCOS operations doesn't currently use bootc at all, would that integration be needed here?